server/lib/models/topic.go

```diff
@@ -10,6 +18,147 @@ type Topic struct {
 }
 
 func InitTopic() {
-	create_table(Topic{}, "topics", "Id")
+	t := create_table(Topic{}, "topics", "Id")
+	t.AddIndex("locale", "INDEX", []string{"Locale"})
+}
+
+const TOPIC_FETCH_LIMIT = 50
+
+func NewTopic(dbif Dbif, a *Account, req *proto.TopicCreateRequest) (*Topic, error) {
+	var (
+		p *Persona
+		c *Channel
+		err error
+	)
+	if p, err = a.Persona(dbif, req.Channel); err != nil {
```

Comment:
__[gometalinter]__
* [gotype] invalid operation: req (variable of type *invalid type) has no field or method Channel